src/main/java/net/earthcomputer/clientcommands/command/StartupCommand.java

```diff
@@ -0,0 +1,91 @@
+package net.earthcomputer.clientcommands.command;
+
+import com.mojang.brigadier.Command;
+import com.mojang.brigadier.CommandDispatcher;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
+import com.mojang.logging.LogUtils;
+import net.earthcomputer.clientcommands.features.ClientCommandFunctions;
+import net.fabricmc.fabric.api.client.command.v2.FabricClientCommandSource;
+import net.minecraft.text.Text;
+import net.minecraft.util.Util;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+
+import static com.mojang.brigadier.arguments.StringArgumentType.*;
+import static net.fabricmc.fabric.api.client.command.v2.ClientCommandManager.*;
+
+public class StartupCommand {
+    private static final Logger LOGGER = LogUtils.getLogger();
+
+    private static final SimpleCommandExceptionType COULD_NOT_CREATE_FILE_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("commands.cstartup.couldNotCreateFile"));
+    private static final SimpleCommandExceptionType COULD_NOT_EDIT_EXCEPTION = new SimpleCommandExceptionType(Text.translatable("commands.cstartup.couldNotEdit"));
+
+    public static void register(CommandDispatcher<FabricClientCommandSource> dispatcher) {
+        dispatcher.register(literal("cstartup")
+            .then(literal("local")
+                .then(literal("add")
+                    .then(argument("command", greedyString())
+                        .executes(ctx -> addStartupCommand(ctx.getSource(), false, getString(ctx, "command")))))
+                .then(literal("edit")
+                    .executes(ctx -> editStartupCommand(false))))
+            .then(literal("global")
+                .then(literal("add")
+                    .then(argument("command", greedyString())
+                        .executes(ctx -> addStartupCommand(ctx.getSource(), true, getString(ctx, "command")))))
+                .then(literal("edit")
+                    .executes(ctx -> editStartupCommand(true)))));
+    }
+
+    private static int addStartupCommand(FabricClientCommandSource source, boolean global, String command) throws CommandSyntaxException {
+        ensureStartupFileExists(global);
+
+        Path file = global ? ClientCommandFunctions.getGlobalStartupFunction() : ClientCommandFunctions.getLocalStartupFunction();
+        if (file == null) {
+            throw COULD_NOT_CREATE_FILE_EXCEPTION.create();
```

Comment:
No file is being created here, right?

Reply:
If `file` is null, that means that it was impossible to create the file a few lines above.